src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/BrokeredMessageToStringConverter.cs

```diff
@@ -2,54 +2,76 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Globalization;
 using System.IO;
+using System.Runtime.Serialization;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml;
 using Microsoft.Azure.WebJobs.Host.Converters;
 using Microsoft.ServiceBus.Messaging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Triggers
 {
     internal class BrokeredMessageToStringConverter : IAsyncConverter<BrokeredMessage, string>
     {
-        public Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
+        public async Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
         {
             if (input == null)
             {
                 throw new ArgumentNullException("input");
             }
+            Stream stream = input.GetBody<Stream>();
+            if (stream == null)
+            {
+                return null;
+            }
 
-            if (input.ContentType == ContentTypes.TextPlain ||
-                input.ContentType == ContentTypes.ApplicationOctetStream ||
-                input.ContentType == ContentTypes.ApplicationJson)
+            TextReader reader = new StreamReader(stream, StrictEncodings.Utf8);
+            try
             {
-                Stream stream = input.GetBody<Stream>();
-                if (stream == null)
+                cancellationToken.ThrowIfCancellationRequested();
+                try
                 {
-                    return Task.FromResult<string>(null);
+                    return await reader.ReadToEndAsync();
+                }
+                catch (DecoderFallbackException)
+                {
+                    // we'll try again below
```

Comment:
This is a bit confusing:
1. If the body is read once, then can we safely do a 2nd attempted read if line 37 throws? Or do we need to buffer the contents?
2. TextReader.ReadToEnd  isn't documented to throw DecoderFallbackException. (https://msdn.microsoft.com/en-us/library/system.io.textreader.readtoendasync(v=vs.110).aspx ) Where does that come from?